### DISCUSSION TITLE
[New Feature Request] Nested Tag​

### DISCUSSION THREAD

**Discussion by houshasen** [toeverything/AFFiNE#6958] [General]
### Description

I appreciate Affine's support for hierarchical note organization, which offers a workflow similar to Notion's subpage creation. Additionally, I believe that "nested tags" can provide another dimension of hierarchical organization, which may be more intuitive in certain situations.

### Detail:

Nested tags, such as #PKM/AFFiNE, can provide a real hierarchical structure to the tags. They are also useful for creating context-dependent tags. In the proper use setting, these functionality would not be possible to reproduce by simple use of multiple tags i.e. #PKM, #AFFiNE. 

**Structural benefit:** One reason to adopt a nested tag system is to create a "true" hierarchy, such as #PKM/AFFiNE and #PKM/Notion. The #PKM tag can have sub-tags, and these can be nested as deeply as desired.

This approach allows for organizing documents in a complete hierarchy under the "tag" section. If implemented, it could also be extended to the "collection" section.

The use case for this would be when there is natural hierachy such as taxonomy in biology. 

**Contextual benefit:** The second benefit or utility case arises when the meaning of certain tags is contextual. Let's consider #Fruit/Apple and #Technology/Apple. 
In this scenario, nested tags are beneficial over the double tag because the tag "Apple" means completely different between two contexts. 

If we used double tags, #Apple will return a mix of both fruit and technology, which most people find unhelpful. Thus, we would end up searching for all documents containing #Software AND #Apple every time tech information is needed.

### Potential Implementation
✅**Storing nested tag**
The entire path should be stored as is i.e. "#PKM/AFFiNE" rather than splitting it into "#PKM" and "#AFFiNE". This is important because the actual path, including the order, could be significant when nested tags are properly used. For example, in taxonomy, we do not want to change the order of Genus/Species, as the correct sequence is crucial. So storing nested tag should not require any extra coding.

⬜**Searching tag**
When searching for a tag such as "AFFiNE", users should be able to see results such as #AFFiNE, #PKM/AFFiNE, and #software/AFFiNE, as all these tags contain the term "AFFiNE".

However, users need to select the specific tag they want from the list. For example, if a user chooses #PKM/AFFiNE, the search results will be based on that particular tag.

I assume this would be a straightforward "text search." I tried to test this capability to see if the current search supports it, but I could not find any free "tag" search zone in AFFiNE at the moment.

⬜**Display tag**
I think this would the main part of programming. These tags should be displayed in a tree view format. This tree structure display be then shown in all appropriate sections such as the "All Docs tag section".

⬜🔥 **Collection upgrade**: Once the basic functions are implemented, they can be used to expand the hierarchy structure within the AFFiNE system, such as in "tag nodes in collections".

### Use case

Offers true hierarchy system, even potentially to collection if tags views are incorporated there. 

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

[a comment by an automated account is not shown]

**Comment by CatsJuice** [comment]
@hwangdev97 cc

**Comment by dsent** [comment]
Hi guys,

Just wanted to say that I'd really want to see this implemented. I'm now trying to move my notes from both Notion and Obsidian, and nested tags is an important part of my structure in Obsidian. I think it can be important for other people moving from Obsidian too.
